### PR TITLE
Add Data.Functor.Classes instances for CofreeF

### PR DESCRIPTION
**Pull request by dbeacham**

### CODE REVIEW

File: src/Control/Comonad/Trans/Cofree.hs [hunk 1]
```diff
@@ -67,6 +69,74 @@ data CofreeF f a b = a :< f b
 #endif
            )
 
+#ifdef LIFTED_FUNCTOR_CLASSES
+instance Show1 f => Show2 (CofreeF f) where
+  liftShowsPrec2 spa _sla spb slb d (a :< fb) =
+    showParen (d > 5) $
+      spa 6 a . showString " :< " . liftShowsPrec spb slb 5 fb
```
Comment (RyanGlScott) [inline src/Control/Comonad/Trans/Cofree.hs]:
```suggestion
      spa 6 a . showString " :< " . liftShowsPrec spb slb 6 fb
```

File: src/Control/Comonad/Trans/Cofree.hs [hunk 2]
```diff
@@ -67,6 +69,74 @@ data CofreeF f a b = a :< f b
...
+#endif
+
+#ifdef LIFTED_FUNCTOR_CLASSES
+instance Read1 f => Read2 (CofreeF f) where
+  liftReadsPrec2 rpa _rla rpb rlb d =
+    readParen (d > 5) $
+      (\r' -> [ (u :< v, w)
+              | (u, s) <- rpa 6 r'
+              , (":<", t) <- lex s
+              , (v, w) <- liftReadsPrec rpb rlb 5 t
```
Comment (RyanGlScott) [inline src/Control/Comonad/Trans/Cofree.hs]:
```suggestion
              , (v, w) <- liftReadsPrec rpb rlb 6 t
```

File: src/Control/Comonad/Trans/Cofree.hs [hunk 3]
```diff
@@ -67,6 +69,74 @@ data CofreeF f a b = a :< f b
...
+instance (Read1 f, Read a) => Read1 (CofreeF f a) where
+  liftReadsPrec = liftReadsPrec2 readsPrec readList
+#else
+instance (Read1 f, Read a) => Read1 (CofreeF f a) where
+  readsPrec1 d =
+    readParen (d > 5) $
+      (\r' -> [ (u :< v,w)
+              | (u, s) <- readsPrec 6 r'
+              , (":<", t) <- lex s
+              , (v, w) <- readsPrec1 5 t
```
Comment (RyanGlScott) [inline src/Control/Comonad/Trans/Cofree.hs]:
```suggestion
              , (v, w) <- readsPrec1 6 t
```

**Comment by dbeacham** [comment]
Good spot! I've added those to the PR.

**Pull request merged**